src/vm/callingconvention.h

```diff
@@ -1276,12 +1289,26 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
     int cbArg = StackElemSize(argSize);
     int cArgSlots = cbArg / STACK_ELEM_SIZE;
 
+    auto setupArgLocDesc = [argType, &thValueType, this](int argOfs)
```

Comment:
It seems that you can do even this the way it is done for amd64. Instead of having this lambda and calling it from three code paths below, it seems you can just add code to initialize the m_argLocDescForStructInRegs inside the `if (thValueType.IsHFA())` under the `case ELEMENT_TYPE_VALUETYPE` above.
At that place,  instead of translating the just generated offset back to the specific m_idxXXXReg, you would just call `m_argLocDescForStructInRegs.Init()` and set the m_argLocDescForStructInRegs.m_idxFloatReg to m_idxFPReg and m_argLocDescForStructInRegs.m_cFloatReg to cFPRegs. Since the lambda below sets the m_argLocDescForStructInRegs for HFA only, it should get the equivalent result.
It looks the change would also be simpler. I hope I have not missed something.